torch/csrc/api/src/nn/modules/activation.cpp

```diff
@@ -38,5 +38,29 @@ void HardshrinkImpl::pretty_print(std::ostream& stream) const {
          << "torch::nn::Hardshrink(" << options.lambda() << ")";
 }
 
+// ============================================================================
+
+HardtanhImpl::HardtanhImpl(const HardtanhOptions& options_)
+    : options(options_) {
+  TORCH_CHECK(options.max_val() > options.min_val(),
+              "max_val must be greater than min_val");
```

Comment:
Copying from our discussion: I think ideally we should put the check in module reset(), with the expectation that in the future we will support the use case of changing a module’s options and then call reset() to reinitialize the module (without calling its constructor)

Reply:
We need to call reset() from the constructor, if we make that change :)

Reply:
fixed